### PR TITLE
Flat list for Restaurants Orders and Dishes

### PR DESCRIPTION
**Pull request by po4erk**

### CODE REVIEW

File: src/services/api/apiAddresses.js [hunk 1]
```diff
@@ -12,7 +12,7 @@ export default {
   USERS: `${API_PATH}/users/`,
   USER_GET: `${API_PATH}/auth/login`,
 
-  RESTAURANTS_GET_ALL: `${API_PATH}/restaurants`,
+  RESTAURANTS_GET_ALL: length => `${API_PATH}/restaurants?limit=3&skip=${length}`,
```
[the remark this answers is not in the document]
Comment (po4erk) [reply src/services/api/apiAddresses.js]: fixed

File: src/reducers/restaurants/reducer.js [hunk 2]
```diff
@@ -3,12 +3,11 @@ import Types from './types'
...
 const initialState = {
   items: [],
-  loading: false,
   error: null,
 }
 
 export default createReducer(initialState, {
-  [Types.GET_RESTAURANTS_SUCCESS]: (state, { payload }) => ({ ...state, items: payload.items, loading: false }),
+  [Types.GET_RESTAURANTS_SUCCESS]: (state, { payload }) => ({ ...state, items: [...payload]}),
   [Types.GET_RESTAURANTS_START]: (state, { payload }) => ({ ...state, loading: true }),
```
Comment (olbesp) [inline src/reducers/restaurants/reducer.js]: Do you need 'loading: true' if you've removed 'loading' from initialState?
You also need to clean selectors of the restaurants' reducer.

File: src/screens/OrderHistory/index.js [hunk 3]
```diff
@@ -44,68 +40,77 @@ class OrderHistory extends Component {
...
+    this.props.getOrders()
   }
 
-  _onRefresh = () => {
-    this.setState({refreshing: true});
-    this.props.getRestaurants()
-    this.props.getOrders().then(() => {
-      this.setState({refreshing: false});
-    });
+  onRefresh = () => {
```
Comment (olbesp) [inline src/screens/OrderHistory/index.js]: Remove it if you don't invoke this method anywhere

File: src/screens/OrderHistory/styles.js [hunk 4]
```diff
@@ -3,14 +3,14 @@ import { StyleSheet } from 'react-native'
...
     paddingTop: 10,
   },
-  ordersContainer: {
-    flex: 1,
-    alignItems: 'center',
-  }
+  dotsContainer: {
+		justifyContent: 'center',
+		alignItems: 'center',
+		paddingVertical: 10,
```
Comment (olbesp) [inline src/screens/OrderHistory/styles.js]: Identation

Comment (po4erk) [reply src/screens/OrderHistory/styles.js]: what do you mean?)

File: src/reducers/dishes/actions.js [hunk 5]
```diff
@@ -1,12 +1,13 @@
 import { getAllDishes as apiGetAllDishes } from '/services/api'
 import Types from './types'
 
-export const getAllDishes = (id) => (dispatch) => {
+export const getAllDishes = (id, list, limit, skip) => (dispatch) => {
   dispatch({ type: Types.GET_DISHES_START, payload: {} })
 
-  return apiGetAllDishes(id)
+  let params = {params: { limit: limit ? limit : 1, skip }}
```
Comment (iarickvigasi) [inline src/reducers/dishes/actions.js]: use default value for argument

File: src/reducers/orders/actions.js [hunk 6]
```diff
@@ -13,14 +13,15 @@ export const sendOrder = (data) => (dispatch) => {
   })
 }
 
-export const getOrders = (data) => (dispatch, getState) => {
-  dispatch({ type: Types.GET_ORDERS_START, payload: data })
+export const getOrders = (list, limit, skip) => (dispatch, getState) => {
+  dispatch({ type: Types.GET_ORDERS_START, payload: {} })
 
   const userId = getState().session.user._id
+  let params = {params: { limit: limit ? limit : 1, skip }}
```
Comment (iarickvigasi) [inline src/reducers/orders/actions.js]: use default arg

File: src/reducers/restaurants/actions.js [hunk 7]
```diff
@@ -1,12 +1,12 @@
 import { getAllRestaurants as apiGetAllRestaurants } from '/services/api'
 import Types from './types'
 
-export const getAllRestaurants = () => (dispatch) => {
+export const getAllRestaurants = (list, limit, skip) => (dispatch) => {
   dispatch({ type: Types.GET_RESTAURANTS_START, payload: {} })
-
-  return apiGetAllRestaurants()
+  let params = {params: { limit: limit ? limit : 1, skip }}
```
Comment (iarickvigasi) [inline src/reducers/restaurants/actions.js]: default arg

File: src/reducers/restaurants/actions.js [hunk 8]
```diff
@@ -1,12 +1,12 @@
 import { getAllRestaurants as apiGetAllRestaurants } from '/services/api'
 import Types from './types'
 
-export const getAllRestaurants = () => (dispatch) => {
+export const getAllRestaurants = (list, limit, skip) => (dispatch) => {
```
Comment (iarickvigasi) [inline src/reducers/restaurants/actions.js]: rename it is not all restaurants anymore

File: src/screens/OrderHistory/index.js [hunk 9]
```diff
@@ -44,68 +40,73 @@ class OrderHistory extends Component {
...
 
-  _onRefresh = () => {
-    this.setState({refreshing: true});
-    this.props.getRestaurants()
-    this.props.getOrders().then(() => {
-      this.setState({refreshing: false});
-    });
+  onScroll = () => {
+    let skip = this.props.orders.length
+		if (this.counter >= skip) { return false }
```
Comment (iarickvigasi) [inline src/screens/OrderHistory/index.js]: why so strange indentation?

File: src/screens/Restaurant/index.js [hunk 10]
```diff
@@ -149,33 +150,58 @@ class Restaurant extends React.Component<Props, State> {
 		this.props.navigation.navigate('Restaurants')
 	}
 
+	onScroll = () => {
+		let skip = this.props.dishes.length
+			if (this.counter >= skip) { return false }
```
Comment (iarickvigasi) [inline src/screens/Restaurant/index.js]: why so strange indentation?

File: src/screens/OrderHistory/index.js (hunk 9 above)
Comment (po4erk) [reply src/screens/OrderHistory/index.js]: github only

**Comment by po4erk** [comment]
should be merge after #59

**Pull request merged**